hashicorp/consul
# [OSS] xDS updates for peerings control plane through mesh gw

### Description
Backport of ENT-3058

We want to enable peering control plane traffic through mesh gateways. This PR updates the xDS generation for OUTBOUND peering Establish connections.  We want to create a route for each unique server cluster, which is identified by `ServerName` in the peering. We also want to make sure the operator has enabled `PeerThroughMeshGateway` in the Mesh Config Entry.


### Testing & Reproduction steps
* Unit Tests added
* Manual Testing
  * Start two clusters
  * Register Mesh Gateway in the default namespace in cluster 1
  * Add Mesh CRD with `PeerThroughMeshGateway` in cluster 1
  * Generate a peering token in cluster 2
  * Establish from 1
  * Even though the gateway isn't receiving traffic, the xDS updates should be observed.


### PR Checklist
